### PR TITLE
[Inference] support llama3 a8w8c8_fp8 inference and cutlass_fp8_gemm

### PR DESCRIPTION
**Pull request by ckl117**
### PR types

New features
### PR changes

Others
### Description

support llama3 a8w8c8_fp8 inference and fp8 cutlass custom op

以下三个文件没有改动，通过copy添加，已明确license：
csrc/gpu/cutlass_kernels/fp8_gemm_fused/dual_gemm/threadblock/dual_mma_base.h
csrc/gpu/cutlass_kernels/fp8_gemm_fused/dual_gemm/threadblock/dual_mma_multistage.h
csrc/gpu/cutlass_kernels/fp8_gemm_fused/dual_gemm/dual_gemm_common.h

### CODE REVIEW

File: llm/predict/predictor.py [hunk 1]
```diff
@@ -85,7 +85,9 @@ class PredictorArgument:
     inference_model: bool = field(default=False, metadata={"help": "whether use InferenceModel to do generation"})
     quant_type: str = field(
         default="",
-        metadata={"help": "Quantization type. Supported values: a8w8, a8w8c8, weight_only_int4, weight_only_int8"},
+        metadata={
+            "help": "Quantization type. Supported values: a8w8, a8w8c8, a8w8_fp8, weight_only_int4, weight_only_int8"
+        },
```
Comment (yuanlehome) [inline llm/predict/predictor.py]: 我的建议是加一个--use_fp8的选项

Comment (ckl117) [reply llm/predict/predictor.py]: 增加--use_fp8后，需要在a8w8后面增加一层判断，代码逻辑就复杂了

File: csrc/gpu/cutlass_kernels/fp8_gemm_fused/dual_gemm/device/dual_gemm.h [hunk 2]
```diff
@@ -0,0 +1,543 @@
+/***************************************************************************************************
```
Comment (DrownFish19) [inline csrc/gpu/cutlass_kernels/fp8_gemm_fused/dual_gemm/device/dual_gemm.h:1]: 这个文件是copy过来的吗？

Comment (ckl117) [reply csrc/gpu/cutlass_kernels/fp8_gemm_fused/dual_gemm/device/dual_gemm.h:1]: 做了一点改动，单独指定了ElementD_参数。
只有以下三个文件没有改动：
csrc/gpu/cutlass_kernels/fp8_gemm_fused/dual_gemm/threadblock/dual_mma_base.h
csrc/gpu/cutlass_kernels/fp8_gemm_fused/dual_gemm/threadblock/dual_mma_multistage.h
csrc/gpu/cutlass_kernels/fp8_gemm_fused/dual_gemm/dual_gemm_common.h

**Pull request merged**